src/Accounts/Authentication/Sanitizer/Services/DefaultSanitizerService.cs

```diff
@@ -39,66 +45,12 @@ internal class DefaultSanitizerService : ISanitizerService
             };
         }
 
-        private static readonly IEnumerable<string> SensitiveDataPatterns = new List<string>()
-        {
-            // AAD client app, most recent two versions.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z-_~.]{3}7Q~[0-9A-Za-z-_~.]{31}\b|\b[0-9A-Za-z-_~.]{3}8Q~[0-9A-Za-z-_~.]{34}" // match
-          + @"\b", // post-match
-            
-            // Prominent Azure provider 512-bit symmetric keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z+/]{76}(APIM|ACDb|\+(ABa|AMC|ASt))[0-9A-Za-z+/]{5}[AQgw]==" // match
-          + @"", // post-match
-
-            // Prominent Azure provider 256-bit symmetric keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z+/]{33}(AIoT|\+(ASb|AEh|ARm))[A-P][0-9A-Za-z+/]{5}=" // match
-          + @"", // post-match
-            
-            // Azure Function key.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z_\-]{44}AzFu[0-9A-Za-z\-_]{5}[AQgw]==" // match
-          + @"", // post-match
-
-            // Azure Search keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z]{42}AzSe[A-D][0-9A-Za-z]{5}" // match
-          + @"\b", // post-match
-            
-            // Azure Container Registry keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z+/]{42}\+ACR[A-D][0-9A-Za-z+/]{5}" // match
-          + @"\b", // post-match
-            
-            // Azure Cache for Redis keys.
-            @"\b" // pre-match
-          + @"[0-9A-Za-z]{33}AzCa[A-P][0-9A-Za-z]{5}=" // match
-          + @"", // post-match
-            
-            // NuGet API keys.
-            @"\b" // pre-match
-          + @"oy2[a-p][0-9a-z]{15}[aq][0-9a-z]{11}[eu][bdfhjlnprtvxz357][a-p][0-9a-z]{11}[aeimquy4]" // match
-          + @"\b", // post-match
-            
-            // NPM author keys.
-            @"\b" // pre-match
-          + @"npm_[0-9A-Za-z]{36}" // match
-          + @"\b", // post-match
-        };
-
         public bool TrySanitizeData(string data, out string sanitizedData)
         {
             if (!string.IsNullOrWhiteSpace(data))
             {
-                foreach (var pattern in SensitiveDataPatterns)
-                {
-                    if (Regex.IsMatch(data, pattern))
-                    {
-                        sanitizedData = Regex.Replace(data, pattern, SanitizedValue);
-                        return true;
-                    }
-                }
+                sanitizedData = this.secretMasker.MaskSecrets(data);
```

Comment:
@michaelcfanning Currently we just need to detect secrets instead of masking it. Could we replace with `this.secretMasker.DetectSecrets` ?

Reply:
@michaelcfanning The function has an out variable for sanitized data. However, in the current phase, this value won't be used and we only use this function to identify if the data has secrets. We will implement the real redaction in the next phase. Since the MaskSecrets has the logging system, I believe it is OK to continue using this function.
In addition, could you please help take a look at the test issue and upgrade the package version to a better one?